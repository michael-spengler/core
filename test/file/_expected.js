module.exports = {
  unit1: [
    {
      function: 'forByte',
      description: 'Increment by 0x11 from 0x00 to 0xee',
      options: {
        initial: 0x00,
        condition: i => i < 0xff,
        increment: i => i + 0x11
      },
      result: [238, 238, 238, 238, 238, 238, 238, 238, 238, 238]
    },
    {
      function: 'forByte',
      description: 'Increment by 0x10 from 0x00 to 0x45: should not increment past 0x40',
      options: {
        initial: 0x00,
        condition: i => i < 0x45,
        increment: i => i + 0x10
      },
      result: [64, 64, 64, 64, 64, 64, 64, 64, 64, 64]
    }
  ]
}
