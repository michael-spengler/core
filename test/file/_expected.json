{
  "unit1": [
    {
      "function": "zeros",
      "description": "Write zeroes: 0x00",
      "result": [0,0,0,0,0,0,0,0,0,0]
    },
    {
      "function": "ones",
      "description": "Write ones: 0xff",
      "result": [255,255,255,255,255,255,255,255,255,255]
    },
    {
      "function": "complementary",
      "description": "Write binary complement",
      "result": [250,5,149,156,31,209,21,109,154,6]
    },
    {
      "function": "byte",
      "description": "Write a byte: 0x22",
      "options": {
        "data": 34
      },
      "result": [34,34,34,34,34,34,34,34,34,34]
    },
    {
      "function": "byte",
      "description": "Write a byte: 0xab",
      "options": {
        "data": 171
      },
      "result": [171,171,171,171,171,171,171,171,171,171]
    },
    {
      "function": "byteArray",
      "description": "Write bytes array: [0x76, 0x6d, 0x3b]",
      "options": {
        "data": [118,109,59]
      },
      "result": [118,109,59,118,109,59,118,109,59,118]
    }
  ]
}